jsonschema-core.xml

```diff
@@ -2566,6 +2477,138 @@
             </section>
         </section>
 
+        <section title="A Vocabulary for Unevaluated Locations">
+            <t>
+                The purpose of these keywords is to enable schema authors to apply
+                subschemas to array items or object properties that have not been
+                successfully evaluated against any dynamic-scope subschema of any
+                adjacent keywords.
+            </t>
+            <t>
+                These instance items or properties may have been unsuccessfully evaluated
+                against one or more adjacent keyword subschemas, such as when an assertion
+                in a branch of an "anyOf" fails.  Such failed evaluations are not considered
+                to contribute to whether or not the item or property has been evaluated.
+                Only successful evaluations are considered.
```

Comment:
I've been thinking about this section and its phrasing.
It's the first time in the whole document we mention "evaluation", and we don't really define what it means.

People (including myself previously) incorrectly assume that if a location has been evaluated (had something applied to it), then it counts as having been evaluated.

I'm going to riff some ideas...

We say here "successful evaluations", we mean if an application doesn't result in validation errors, and therefore the applied schema or keyword is correctly applicable to the instance location.

For example, if you had a schema...
`anyOf [ { title> unicycle, props > wheels > const > 1 }, { title: bicycle, props > wheels > const > 2 } ]`

If your instance was a unicycle, the first subschema item in the `anyOf` would be applied and _**successfully evaluated**_. Because it only has one wheel, we know it's a unicycle. It is not a bicycle, so that whole subschema is not successfully evaluated.

The instance is successfully evaluated as (or determined to be) a unicycle.

We can further justify this terminology by the fact that evaluation can be short-circuited, and therefore there's no expectation for a schema to be fully evaluated (although that would be a weak argument, and probably confuse things).

---

We need to clearly define what successful evaluation means, and what unsuccessful evaluation means, with justification (IMHO). I'm not currently convinced I can do that.

I'm happy for this to be a new issue in order to get this PR merged, but it would be essential to 2020-NN.

Reply:
This is a good point, and it should be made clear that these keywords really mean "un**Sucessfully**Evaluated*Thing*" rather than just "unEvaluated*Thing*".